src/rce.ts

```diff
@@ -40,45 +49,99 @@ export function handleCancelRequest(_actionData: ActionData): string | undefined
  * Only runs if there is an RCE callback in NEURO.
  */
 export function emergencyDenyRequests(): void {
-    if (!NEURO.rceCallback) {
+    if (!NEURO.rceRequest) {
         return;
     }
     clearRceDialog();
-    logOutput("INFO", `Cancelled ${NEURO.rceCallback} due to emergency shutdown.`)
-    NEURO.client?.sendContext("Your last request was denied.")
-    vscode.window.showInformationMessage("The last request from Neuro has been denied automatically.")
+    logOutput('INFO', `Cancelled ${NEURO.rceRequest.callback} due to emergency shutdown.`);
+    NEURO.client?.sendContext('Your last request was denied.');
+    vscode.window.showInformationMessage('The last request from Neuro has been denied automatically.');
 }
 
 export function clearRceDialog(): void { // Function to clear out RCE dialogs
-    NEURO.rceCallback = null;
+    NEURO.rceRequest?.dismiss();
+    NEURO.rceRequest = null;
     NEURO.client?.unregisterActions([cancelRequestAction.name]);
     NEURO.statusBarItem!.tooltip = 'No active request';
     NEURO.statusBarItem!.color = new vscode.ThemeColor('statusBarItem.foreground');
     NEURO.statusBarItem!.backgroundColor = new vscode.ThemeColor('statusBarItem.background');
 }
 
 export function openRceDialog(): void {
-    if(!NEURO.rceCallback)
+    if(!NEURO.rceRequest)
         return;
 
-    const callback = NEURO.rceCallback;
-    const message = typeof NEURO.statusBarItem!.tooltip === 'string'
-        ? NEURO.statusBarItem!.tooltip
-        : NEURO.statusBarItem!.tooltip!.value;
-    vscode.window.showInformationMessage(message, 'Accept', 'Deny').then(
-        (value) => {
-            if(NEURO.rceCallback !== callback) // Multiple messages may be opened, ensure that callback is only called once
-                return;
-            clearRceDialog();
-            if(value === 'Accept') {
-                NEURO.client?.sendContext('Vedal has accepted your request.');
-                const result = callback();
-                if(result)
-                    NEURO.client?.sendContext(result);
-            } else {
-                NEURO.client?.sendContext('Vedal has denied your request.');
-            }
+    // we can't add any buttons to progress, so we have to add the accept link
+    const message = `${NEURO.rceRequest.prompt} [Accept](command:neuropilot.confirmRceRequest)`;
```

Comment:
If possible can we also have it say Accept or Deny?

```suggestion
    const message = `${NEURO.rceRequest.prompt} [Accept](command:neuropilot.confirmRceRequest), or [Deny](command:neuropilot.declineRceRequest).`;
```

Reply:
Actually the Cancel button works fine idk what I was one when I was typing that. OCD on adding a Deny button maybe?